transcom/mymove
# E-04430-Okta-Profile-Edit

## [Okta - Customer Edits Okta Profile from MilMove](https://www13.v1host.com/USTRANSCOM38/Epic.mvc/Summary?oidToken=Epic%3A830848)

## Summary

When a MilMove Customer registers/signs into MilMove and successfully authenticates the user, they can now see their Okta profile values and update them within the MilMove app itself. An Okta profile form was added to the profile page, a user can click edit and update their Okta profile values and then be returned to the profile page.

This uses an API key (which has been added to chamber) and calls the Okta User API to update their profile, then gets the new profile data and updates the state in Redux.

[Okta Users API](https://developer.okta.com/docs/reference/api/users/) explains the Users API used to get and update the user profile, which is what is being used in this PR.

[Okta Authentication Methods](https://developer.okta.com/docs/reference/core-okta-api/#authentication) goes over authentication methods for OIDC app, which we chose to use an API key because Okta's myAccount API does not support what we were trying to accomplish in this PR and is way more limited compared to the Users API.

## Verification Steps for Reviewers

- Ensure Okta information is accurate on both pre and post-edit.
- If you do not provide an EDIPI to Okta, it should say "Not Provided" in the display box
- On submission, it should redirect you to the profile page and display a success banner
- Validates inputs and displays errors if values are not meeting criteria (email format, EDIPI needs to be between 7-10 digits in length)

### Setup to Run the Code

- pull the branch
- `make client_run`
- `make server_run`

### How to test

1. Log into the app as a customer
2. Confirm your Okta profile values are in the profile page
3. Update profile values and confirm changes were made in state
4. Attempt to validate with different values to confirm form validation is working
